src/test/java/com/google/sps/EventsServletTest.java

```diff
@@ -0,0 +1,209 @@
+package com.google.sps;
+
+import static com.google.sps.utils.StringConstants.*;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.api.client.util.DateTime;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+import com.google.api.services.calendar.model.EventDateTime;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.servlets.EventsServlet;
+import com.google.sps.TestUtils;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+
+@RunWith(JUnit4.class)
+public final class EventsServletTest extends Mockito {
+  //3600000ms = 1 hour
+  private final long TIME_OFFSET = 3600000L;
+  private final String TEST_EVENT_DESCRIPTION = "test event description";
+  private final String TEST_EVENT_HTML = "test event HTML";
+  private final String TEST_EVENT_LOCATION = "test event location";
+  private final String TEST_EVENT_TITLE = "test event title";
+  
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+  private final DateTimeFormatter formatter = DateTimeFormatter
+      .ofPattern("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
+      .withZone(ZoneId.of(TIMEZONE));
+  
+  private String initialEventTime;
+  private String endEventTime;
+
+  private EventsServlet servlet;
+  private DatastoreService datastore;
+  private Entity groupEntity;
+  private PrintWriter printWriter;
+  private StringWriter stringWriter;
+
+  @Mock
+  HttpServletRequest request;
+
+  @Mock
+  HttpServletResponse response;
+
+  @Spy
+  EventsServlet servletSpy;
+
+  @Before
+  public void setUp() throws IOException {
+    servlet = new EventsServlet();
+
+    helper.setUp();
+    datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    stringWriter = new StringWriter();
```

Comment:
this can go in the variable declaration section since it doesn't depend on anything.

Same with the `servlet` variable and also `datastore`

Reply:
For testing purposes, we need to declare the datastore every time since it depends on the helper before it being setUp (has to do with the way to test anything with datastore, we have to do it before every test)